res/controllers/Pioneer-DDJ-SB3-scripts.js

```diff
@@ -0,0 +1,1791 @@
+    PioneerDDJSB3.padForDeck[2] = new PioneerDDJSB3.Pad(2);
+    PioneerDDJSB3.padForDeck[3] = new PioneerDDJSB3.Pad(3);
+    PioneerDDJSB3.padForDeck[4] = new PioneerDDJSB3.Pad(4);
+
+    PioneerDDJSB3.bindNonDeckControlConnections(false);
+    PioneerDDJSB3.initDeck("[Channel1]");
+    PioneerDDJSB3.initDeck("[Channel2]");
+    PioneerDDJSB3.initDeck("[Channel3]");
+    PioneerDDJSB3.initDeck("[Channel4]");
+
+    if (PioneerDDJSB3.twinkleVumeterAutodjOn) {
+        PioneerDDJSB3.vuMeterTimer = engine.beginTimer(100, "PioneerDDJSB3.vuMeterTwinkle()");
+    }
+
+    // request the positions of the knobs and faders from the controller
+    midi.sendShortMsg(0x9B, 0x09, 0x7f);
+
+    PioneerDDJSB3.flasher.init();
+    PioneerDDJSB3.initFlashingPadLedControl();
+};
+
+PioneerDDJSB3.shiftListeners = [];
+
+PioneerDDJSB3.Deck = function(deckNumber) {
+    var theDeck = this;
+    this.group = "[Channel" + deckNumber + "]";
+
+    this.shiftButton = function(channel, control, value, status, group) {
+        if (value > 0) {
+            theDeck.shift();
+            PioneerDDJSB3.shiftPressed = true;
+            PioneerDDJSB3.chFaderStart[deckNumber] = null;
+            PioneerDDJSB3.headphoneLevel.shift();
+            for (var i = 0; i < PioneerDDJSB3.shiftListeners.length; i++) {
+                PioneerDDJSB3.shiftListeners[i](group, true);
+            }
+        } else {
+            theDeck.unshift();
+            PioneerDDJSB3.shiftPressed = false;
+            PioneerDDJSB3.headphoneLevel.unshift();
+            // eslint-disable-next-line no-redeclare
+            for (var i = 0; i < PioneerDDJSB3.shiftListeners.length; i++) {
+                PioneerDDJSB3.shiftListeners[i](group, false);
+            }
+        }
+    };
+    var playButton = this.playButton = new components.PlayButton({
+        midi: [0x90 + deckNumber - 1, 0x0B],
+        shiftOffset: 60,
+        shiftControl: true,
+        sendShifted: true,
+        connect: function() {
+            this.connections[0] = engine.makeConnection(this.group, "play_indicator", this.output);
+            this.connections[1] = engine.makeConnection(this.group, "track_loaded", this.output);
+        },
+        flashLed: function(active) {
+            playButton.send(playButton.outValueScale(active));
+        },
+        output: function(value, group, control) {
+            playButton[control] = value;
+
+            if (playButton.track_loaded) {
+                if (playButton.play_indicator) {
+                    PioneerDDJSB3.flasher.removeFunction(playButton.flashLed);
+                    playButton.send(playButton.outValueScale(true));
+                } else {
+                    PioneerDDJSB3.flasher.addFunction(playButton.flashLed);
+                }
+            } else {
+                PioneerDDJSB3.flasher.removeFunction(playButton.flashLed);
+                playButton.send(playButton.outValueScale(false));
+            }
+        }
+    });
```

Comment:
This flashes the play button when a track is loaded but isn't playing. It can work without the flashing. I can remove the flashing bits if that is the preferred behavior.

Reply:
Flashing should be used sparingly only for temporary states. Using flashing for a long lasting and commonly occurring state is annoying. Please remove that.